### PR TITLE
Process script updates one by one

### PR DESCRIPTION
**Pull request by da-kami**
Previously we assembled the complete history of all batches and then processed all updates.
With this change we now process them one by one, update the state and only accumulate the events that are ready.

The state update is cheap and can be done as part of processing the request.
Handling the CFD update could be done outside of sync, but we could regard that update as part of sync.
Only once the CFD is updated the sync is actually complete, because only then the state is communicated in the system, so it might be better to keep that within this scope.

### CODE REVIEW

File: crates/daemon/src/monitor.rs [hunk 1]
```diff
@@ -1012,9 +957,76 @@ struct ReinitMonitoring {
...
+                            height: history.height,
+                            tx_hash: history.tx_hash,
+                        })
+                        .collect();
+
+                    let mut ready_events = self.state.update(latest_block_height, script_status);
+
+                    if ready_events.is_empty() {
+                        return;
+                    }
```
Comment (luckysori) [inline crates/daemon/src/monitor.rs]: ❓ Is this necessary?

**Review by luckysori — APPROVED** [review]

**Comment by da-kami** [comment]
@luckysori I did another iteration on this, because I was not convinced anymore that processing the CFD update should not be in scope of sync. I don't think we have to trigger another sync while we are still updating the CFDs with the result of the previous sync.

I found a simpler solution for optimizing the memory consumption further, while not moving the CFD update outside, I think it's good enough to aim for that. The changeset is smaller.

**Comment by da-kami** [comment]
Looks like this might not actually an optimization, see: https://github.com/itchysats/itchysats/pull/3197#issuecomment-1282219467

**Comment by da-kami** [comment]
Closing this for now, because we don't see an actual performance improvement by this code.

**Comment by bonomat** [comment]
If you don't need the branch anymore, can you delete it please?

**Pull request closed without merging**